dlt/common/destination/utils.py

```diff
@@ -210,7 +216,11 @@ def prepare_load_table(
         prep_table = fill_hints_from_parent_and_clone_table(tables, table)
         table_format = prep_table.get("table_format")
         # set table format if preferred
-        if destination_capabilities.preferred_table_format and not table_format:
+        if (
```

Comment:
this is a good fix! but should be isolated to only `filesystem` destination. all other destinations follow (or will follow) the table format. look for `prepare_load_table` in `filesystem.py`. all destinations can customize how tables are prepared.